### PR TITLE
feat: tutorial examples

### PR DESCRIPTION
**Pull request by fjogeleit**
* add playground examples for the different type of rules and features like variable mocking
* add instruction dialog

Bildschirmfoto 2023-05-14 um 10 50 24
Bildschirmfoto 2023-05-14 um 11 01 57
Bildschirmfoto 2023-05-14 um 11 02 42

### CODE REVIEW

File: frontend/src/components/HelpButton.vue [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+          <v-btn @click="dialog = false" icon="mdi-close"></v-btn>
+        </template>
+      </v-toolbar>
+      <v-card-text>
+        <h2 class="text-h6 mb-2 mt-3">This Playground supports the following kinds of rules</h2>
+
+        <ul class="pl-4 my-2">
+          <li>Validation Rules</li>
+          <li>Mutation Rules</li>
+          <li>Geneneration Rules</li>
```
Comment (chipzoller) [inline frontend/src/components/HelpButton.vue:26]: Misspelled

File: frontend/src/components/HelpButton.vue [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+          <li>ImageVerification Rules</li>
```
Comment (chipzoller) [inline frontend/src/components/HelpButton.vue:27]: Space between two words.

File: frontend/src/components/HelpButton.vue [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        </ul>
+
+        <p>
+          Not possible is to access private registries for e.g. ImageVerification rules, also not supported is the "clone" functionality of generate rules.
```
Comment (chipzoller) [inline frontend/src/components/HelpButton.vue:31]: "It is not possible to access private registries (e.g., for image verification rules) or to use the clone capability in a generate rule."

File: frontend/src/components/HelpButton.vue [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+        </p>
+
+        <h2 class="text-h6 mb-2 mt-3">Context and Variables</h2>
+
+        <p>
+          Currently it is not possible to add variables from resources
+          or do actual APICalls, instead its possible to mock variables over the
+          "variables" configuration in the "context" input
```
Comment (chipzoller) [inline frontend/src/components/HelpButton.vue:39]: "It is currently not possible to add variables from external resources or do actual API calls. It is only possible to mock variables using the `variables` configuration in the context input."

File: frontend/src/components/HelpButton.vue [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+        </p>
+
+        <h2 class="text-h6 mb-2 mt-3">Multiple Manifests</h2>
+
+        <p>
+          It is supported to define multiple policies and/or resources as input,
+          the context as well as variables will be shared and used for all executations.
```
Comment (chipzoller) [inline frontend/src/components/HelpButton.vue:46]: "It is supported to define multiple policies and/or resources as inputs. Context and variables will be shared for all executions."

File: frontend/src/components/HelpButton.vue [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+        </p>
+
+        <h2 class="text-h6 mb-2 mt-3">Load Manifests</h2>
+
+        <ul class="pl-4 my-2 text-body-1">
+          <li>The "File" Button loads a local YAML file as input</li>
+          <li>The "URL" Button load an manifest from an external URL<br />Example: <a href="https://raw.githubusercontent.com/kyverno/policies/main/best-practices/disallow-latest-tag/disallow-latest-tag.yaml" target="_blank">https://raw.githubusercontent.com/kyverno/policies/main/best-practices/disallow-latest-tag/disallow-latest-tag.yaml</a></li>
```
Comment (chipzoller) [inline frontend/src/components/HelpButton.vue:53]: "loads a manifest"

File: frontend/src/config.ts [hunk 7]
```diff
@@ -34,8 +35,21 @@ export const options = {
         { name: 'HC Light', theme: 'hc-light' }
     ],
     examples: {
+        "Tutorials": {
+            path: '/tutorials',
+            color: 'warning',
+            policies: [
+                { folder: 'validate', title: 'Validation' },
+                { folder: 'mutation', title: 'Mutation' },
+                { folder: 'image-verify', title: 'ImageVerify' },
```
Comment (chipzoller) [inline frontend/src/config.ts:44]: The rule type is `verifyImages` so that's what should be used for an organizer.

**Review by chipzoller — CHANGES_REQUESTED** [review]
In general, this looks fine, but there's really only one example here which makes use of variables and it's very simple. So I'm confused if the objective is to add simply more examples or specifically examples for a tutorial which illustrates how to use different contexts and variables. Nothing here really demonstrates these capabilities well if that's the case.

**Comment by fjogeleit** [comment]
Hey Chip. thanks for your feedback. You're right, the current examples are very basic and cover not all possibilities and features. They are more a starting point to show case the basic Idea of tutorials.

at this point I am somewhat dependent on contributions because I have worked little with custom rules and do not know all the possibilities. In principle, I would like to show more examples here or elaborate on the existing ones.

**Comment by chipzoller** [comment]
I will provide some suggestions today.

**Comment by chipzoller** [comment]
## ConfigMap Context Examples

* https://kyverno.io/policies/other/allowed_pod_priorities/allowed_pod_priorities/
* https://main.kyverno.io/policies/other/exclude-namespaces-dynamically/exclude-namespaces-dynamically/

## API Call Context Examples

* https://main.kyverno.io/policies/other/restrict-pod-count-per-node/restrict-pod-count-per-node/ (good easy one)
* https://main.kyverno.io/policies/other/restrict-ingress-host/restrict-ingress-host/
* https://main.kyverno.io/policies/other/require-netpol/require-netpol/

## UPDATE Examples

* https://kyverno.io/policies/other/allowed_label_changes/allowed-label-changes/
* https://main.kyverno.io/policies/other/block-updates-deletes/block-updates-deletes/ (Also uses context for clusterrole)

## CONNECT Examples

* https://main.kyverno.io/policies/other/block-pod-exec-by-namespace-label/block-pod-exec-by-namespace-label/

## Subject Examples

* https://main.kyverno.io/policies/other/check-serviceaccount/check-serviceaccount/


These should be good to get started.

**Pull request merged**